Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -0,0 +1,1076 @@
+        /// </summary>
+        private List<Hyperlink> _listeningHyperlinks = new List<Hyperlink>();
+
+        /// <summary>
+        /// The root element for our rendering.
+        /// </summary>
+        private Border _rootElement = null;
+
+        /// <summary>
+        /// Fired when the text is done parsing and formatting. Fires each time the markdown is rendered.
+        /// </summary>
+        public event EventHandler<OnMarkdownRenderedArgs> OnMarkdownRendered;
+
+        /// <summary>
+        /// Fired when a link element in the markdown was tapped.
+        /// </summary>
+        public event EventHandler<OnLinkClickedArgs> OnLinkClicked;
+
+        /// <summary>
+        /// Gets or sets the markdown text to display.
+        /// </summary>
+        public string Text
+        {
+            get { return (string)GetValue(TextProperty); }
+            set { SetValue(TextProperty, value); }
+        }
+
+        public static readonly DependencyProperty TextProperty = DependencyProperty.Register(
+            nameof(Text),
+            typeof(string),
+            typeof(MarkdownTextBlock),
+            new PropertyMetadata(string.Empty, new PropertyChangedCallback(OnPropertyChangedStatic)));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether text selection is enabled.
+        /// </summary>
+        public bool IsTextSelectionEnabled
+        {
+            get { return (bool)GetValue(IsTextSelectionEnabledProperty); }
+            set { SetValue(IsTextSelectionEnabledProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="IsTextSelectionEnabled"/>.
+        /// </summary>
+        public static readonly DependencyProperty IsTextSelectionEnabledProperty = DependencyProperty.Register(
+            nameof(IsTextSelectionEnabled),
+            typeof(bool),
+            typeof(MarkdownTextBlock),
+            new PropertyMetadata(true, new PropertyChangedCallback(OnPropertyChangedStatic)));
+
+        /// <summary>
+        /// Gets or sets the brush used to fill the background of a code block.
+        /// </summary>
+        public Brush CodeBackground
+        {
+            get { return (Brush)GetValue(CodeBackgroundProperty); }
+            set { SetValue(CodeBackgroundProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="CodeBackground"/>.
+        /// </summary>
+        public static readonly DependencyProperty CodeBackgroundProperty = DependencyProperty.Register(
+            nameof(CodeBackground),
+            typeof(Brush),
+            typeof(MarkdownTextBlock),
+            new PropertyMetadata(new SolidColorBrush(), new PropertyChangedCallback(OnPropertyChangedStatic)));
+
+        /// <summary>
+        /// Gets or sets the brush used to render the border fill of a code block.
+        /// </summary>
+        public Brush CodeBorderBrush
```

Comment:
Lol, I'm still not 100% there. (I'm more of a C++ guy, I'm not super good with XAML).

Are you suggesting I pull all of the properties out into the <Style ...> element defined in the .xaml file? Or make another style that sets them all?